servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -640,4 +648,63 @@ public String toString() {
                     '}';
         }
     }
+
+    private static final class InstrumentedConnectionFactory<Addr, C extends LoadBalancedConnection>
+            extends DelegatingConnectionFactory<Addr, C> {
+
+        private final ConnectTracker connectTracker;
+
+        InstrumentedConnectionFactory(final ConnectionFactory<Addr, C> delegate, ConnectTracker connectTracker) {
+            super(delegate);
+            this.connectTracker = connectTracker;
+        }
+
+        @Override
+        public Single<C> newConnection(Addr addr, @Nullable ContextMap context, @Nullable TransportObserver observer) {
+            return super.newConnection(addr, context, observer).liftSync(delegate -> new ConnectSubscriber(delegate));
+        }
+
+        private class ConnectSubscriber implements SingleSource.Subscriber<C> {
+
+            private final AtomicBoolean once = new AtomicBoolean();
```

Comment:
We made an agreement for our codestyle to always use atomic field updaters when an atomic variable can not be `static`. The goal is to reduce memory allocations without the need to consider each case or worry if frequency of allocations ever increases to start being visible.